cmake/cotire.cmake

```diff
@@ -0,0 +1,4190 @@
+# - cotire (compile time reducer)
```

Comment:
If you don't like having this file in a tree, I can add it as a git submodule from https://github.com/sakra/cotire

Yes, I need it for speeding up MSVC build. On my computer, using of precompiled headers gives a two-fold improvement in build time. Theoretically, the PCH should improve compilation time for other compilers, but in practice, I did not see any significant effects.

PCH does not affect build on Linux, because it is enabled only for MSVC. Most likely, the build time has increased due adding cpp-netlib through `add_subdirectory()`. The subproject inherits all global parent's compile definitions and options. This can be avoided if we add a subproject through [ExternalProject](https://cmake.org/cmake/help/latest/module/ExternalProject.html) instead of `add_subproject()`.

Reply:
>I can add it as a git submodule from https://github.com/sakra/cotire

Sure.